### PR TITLE
Add test for naming conventions

### PR DESCRIPTION
**Pull request by gkinsman**
Add integration test test to ensure naming conventions are applied correctly, and generate the correct queue/topic names

### CODE REVIEW

File: tests/JustSaying.IntegrationTests/Fluent/Configuration/Approvals/WhenUsingNamingConventions.ThenTheNamingConventionIsApplied.approved.txt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      {
+        "MessageType": "SimpleMessage",
+        "QueueName": "beforequeue-simplemessage-afterqueue",
+        "MiddlewareChain": [
+          "MessageContextAccessorMiddleware",
+          "ErrorHandlerMiddleware",
+          "LoggingMiddleware",
+          "StopwatchMiddleware",
+          "SqsPostProcessorMiddleware",
+          "HandlerInvocationMiddleware`1"
```
Comment (martincostello) [inline tests/JustSaying.IntegrationTests/Fluent/Configuration/Approvals/WhenUsingNamingConventions.ThenTheNamingConventionIsApplied.approved.txt:14]: I wonder if these should print out including the type argument's type name?

Comment (gkinsman) [reply tests/JustSaying.IntegrationTests/Fluent/Configuration/Approvals/WhenUsingNamingConventions.ThenTheNamingConventionIsApplied.approved.txt]: Yeah that's a good shout, let me know what you think about the [latest commit](https://github.com/justeat/JustSaying/pull/973/commits/a5f9cb64f7c0e2ef092484d2920c5ce84e874716) 👍

File: src/JustSaying/Extensions/TypeExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+namespace JustSaying.Extensions;
+
+public static class TypeExtensions
```
Comment (martincostello) [inline src/JustSaying/Extensions/TypeExtensions.cs:3]: Does this need to be public?

Comment (gkinsman) [reply src/JustSaying/Extensions/TypeExtensions.cs]: Nope, fixed 👍

**Review by martincostello — APPROVED** [review]

**Pull request merged**